packages/nimble-components/src/banner/specs/README.md

```diff
@@ -0,0 +1,197 @@
+
+When the user presses the dismiss button, we will hide the banner (`display:none`) and fire a `toggle` event.
+
+### API
+
+_The key elements of the component's public API surface:_
+
+-   _Component Name_: `nimble-banner`
+-   _Props/Attrs_:
+    -   `open` - whether the banner is visible or not
+    -   `prevent-dismiss` - set to hide the dismiss button (attr name taken from Nimble dialog)
+    -   `severity` - one of `error`, `warning`, `info`, or undefined (default)
+    -   `title-hidden` - set to hide the title text, which should always be provided for a11y reasons
+    -   `dismiss-button-label` - a localized label to give the dismiss button (for a11y purposes)
+-   _Methods_
+-   _Events_
+    -   `toggle` - fired when the banner is closed or opened. Event has `newState` and `oldState`.
+-   _Slots_
+    -   `title` - for the title/header text
+    -   (default) - for the primary message text
+    -   `action` - for the action button/link
+-   _CSS Classes and CSS Custom Properties that affect the component_
+
+We only formally support spans of text in the `title` and default slots, but we will not explicitly prevent other HTML from being slotted there. The `action` slot only supports a single `nimble-button` or `nimble-anchor`, but again, we will not do anything to enforce this.
+
+### Examples
+
+```html
+<nimble-banner severity="info" dismiss-button-label="Close">
+    <span slot="title">License Expiring</span>
+    Your license will expire at the end of the month. To renew, go to
+    ni.com/renew.
+    <nimble-anchor slot="action" href="...">Renew my license</nimble-anchor>
+</nimble-banner>
+```
+
+```html
+<nimble-banner severity="error" dismiss-button-label="Close">
+    <span slot="title">Authentication Failed</span>
+    Could not authenticate. Please re-enter your credentials from Settings.
+    <nimble-button slot="action">Open Settings</nimble-button>
+</nimble-banner>
+```
+
+```html
+<nimble-banner title-hidden prevent-dismiss>
+    <span slot="title">Demo mode message</span>
+    You are operating in demo mode. Some features will not be available.
+</nimble-banner>
+```
+
+### Anatomy
+
+<!-- prettier-ignore -->
+```html
+<div class="icon">
+    ${when(x => x.severity === 'error', html`
+        <nimble-icon-exclamation-mark></nimble-icon-exclamation-mark>`)}
+    ${when(x => x.severity === 'warning', html`
+        <nimble-icon-triangle-filled></nimble-icon-triangle-filled>`)}
+    ${when(x => x.severity === 'info', html`
+        <nimble-icon-info></nimble-icon-info>`)}
+</div>
+<div class="text">
+    ${when(x => !x.titleHidden, html`
+        <slot name="title"></slot>)}
+    <slot></slot>
+</div>
+<div class="end">
+    <slot name="action"></slot>
+    <div class="close">
+        ${when(x => !x.preventDismiss), html`
+            <nimble-button appearance="ghost" content-hidden>
+                <nimble-icon-xmark slot="start"></nimble-icon-xmark>
```

Comment:
For completeness, you might show how the dismiss button label will be used, which I'm guessing is like this:

```suggestion
                <nimble-icon-xmark slot="start"></nimble-icon-xmark>
                ${x => x.dismissButtonLabel}
```